src/components/ReloadButton.jsx

```diff
@@ -1,48 +1,38 @@
-import React, {
-  useContext, useEffect,
-} from 'react';
-import { Button, CircularProgress } from '@mui/material';
+/* eslint-disable no-useless-concat */
+import React, { useContext } from 'react';
+import { Button } from '@mui/material';
 
-import { useClient, useHandleProcess, useLogger } from 'kumo-app';
+import aruku_interfaces from '../proto/aruku_grpc_web_pb';
 
 import WalkContext from '../context/WalkContext';
 
 function ReloadButton() {
-  const {
-    setKinematicValue, setWalkingValue,
-  } = useContext(WalkContext);
-
-  const client = useClient();
-  const logger = useLogger();
-
-  const [reloading, handleReload] = useHandleProcess(() => client
-    .call({})
-    .then((response) => {
-      logger.success('Successfully get config.');
-      const kinematic = JSON.parse(`${response.json_kinematic.replace('/\\/g', '')}`);
-      const walking = JSON.parse(`${response.json_walking.replace('/\\/g', '')}`);
-      Object.keys(kinematic).map((name) => Object.keys(kinematic[name])
-        .map((key) => setKinematicValue(name, key, kinematic[name][key])));
-      Object.keys(walking).map((name) => Object.keys(walking[name])
-        .map((key) => setWalkingValue(name, key, walking[name][key])));
-    })
-    .catch((err) => {
-      logger.error(`Failed to load config! ${err.message}.`);
-    }), 500);
-
-  useEffect(() => {
-    handleReload();
-  }, []);
+  const { setPublished, setKinematic, setWalking } = useContext(WalkContext);
+
+  const client = new aruku_interfaces.ConfigClient(`${import.meta.env.GRPC_WEB_API_URL}`, null, null);
+  const request = new aruku_interfaces.Empty();
+
+  const handleReload = () => {
+    client.getConfig(request, {}, (err, response) => {
+      if (err) {
+        console.log(`Unexpected error: code = ${err.code}` + `, message = "${err.message}"`);
+      } else {
+        setKinematic(JSON.parse(response.array[0]));
+        setWalking(JSON.parse(response.array[1]));
+      }
+    });
+
+    setPublished(false);
+  };
 
   return (
     <Button
```

Comment:
[optional] To avoid spam requests, you can use LoadingButton (you can disable using the component's props when the data is still loading). please check other buttons too